src/App.tsx

```diff
@@ -1,54 +1,95 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+import classNames from 'classnames';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [isDropDownActive, setIsDropDownActive] = useState(false);
+  const delay = 1000;
```

Comment:
The delay is hardcoded. Consider making it a prop so it can be customizable as described in the task.